postgres/datadog_checks/postgres/metadata.py

```diff
@@ -472,4 +476,10 @@ def _collect_postgres_settings(self):
                 cursor.execute(PG_SETTINGS_QUERY)
                 rows = cursor.fetchall()
                 self._log.debug("Loaded %s rows from pg_settings", len(rows))
-                return [dict(row) for row in rows]
+                return [dict(row) for row in rows if not self._is_ignored_setting(row)]
```

Comment:
can we add a where clause to the `PG_SETTINGS_QUERY` to exclude the ignored settings from the query? 
additionally, instead of prefixes, maybe we can use [patterns matching](https://www.postgresql.org/docs/current/functions-matching.html#FUNCTIONS-MATCHING) to support any user defined patterns?